src/hooks/useCreateTrackMutation.ts

```diff
@@ -0,0 +1,18 @@
+import { useNavigate } from "react-router-dom";
+import { createTrack } from "../api/endpoints";
+import useHandleMutationResult from "./useHandleMutationResult";
+
+export function useCreateTrackMutation() {
```

Comment:
I suggest to move all hooks that are related to certain endpoints in another folder, related to this domain (e.g. like your "endpoint" folder inside "api").
Here you are mixing api-related hooks with general hooks (like for debounce). It creates a little mess